fireapp/lib/presentation/volunteer_home/volunteer_home_widget.dart

```diff
@@ -32,11 +35,108 @@ class _HomeState extends FireAppState<HomeView>
+              print('Exception state: ${exceptionState.exception}');
+              return Center(child: Text('Error: ${exceptionState.exception}'));
+            } else if (state is SuccessRequestState<List<Shift>>) {
+              print('Success state with ${state.result.length} shifts');
+              final shifts = state.result;
+              if (shifts.isEmpty) {
+                return const Center(child: Text("No shifts available."));
+              }
+
+              final nextShift = shifts.first;
+
+              return SingleChildScrollView(
+                  child: Padding(
+                      padding: const EdgeInsets.all(24.0),
+                      child: Column(
+                        crossAxisAlignment: CrossAxisAlignment.start,
+                        children: [
+                          const Text(
+                            'Your Next Shift:',
+                            style: TextStyle(fontSize: 24, fontWeight: FontWeight.w400),
+                          ),
+                          SizedBox(height: 16),
+                          _buildShiftCard(nextShift, isNextShift: true),
+                          const SizedBox(height: 24,),
+                          const Text(
+                            'Upcoming Shifts:',
+                            style: TextStyle(fontSize: 24, fontWeight: FontWeight.w400),
+                          ),
+                          SizedBox(height: 16,),
+                          ListView.builder(
+                            shrinkWrap: true,
+                            physics: const NeverScrollableScrollPhysics(),
+                            itemCount: shifts.length - 1,
+                            itemBuilder: (context, index) {
+                              return Padding(
+                                  padding: const EdgeInsets.only(bottom: 8.0),
+                                  child: _buildShiftCard(shifts[index + 1])
+                              );
+                            },
+                          )
+                        ],
+                      )
+                  )
+              );
+            } else {
+              return const Center(child: Text('Unknown state'));
+            }
+          });
+
+  }
+
+  Widget _buildShiftCard(Shift shift, {bool isNextShift = false}) {
+    final dateFormat = DateFormat('EEEE d MMMM');
+    final timeFormat = DateFormat('h:mm a');
+
+    return Card(
+      color: isNextShift ? Colors.red[200] : Colors.blue[200],
+      child: Padding(
+        padding: const EdgeInsets.all(16.0),
+        child: Column(
+          crossAxisAlignment: CrossAxisAlignment.start,
+          children: [
+            Row(
+              mainAxisAlignment: MainAxisAlignment.spaceBetween,
+              children: [
+                Text(dateFormat.format(shift.startTime),
+                      style: TextStyle(fontWeight: FontWeight.bold)),
+                Text('Shift ID: ${shift.shiftId}')
+              ],
```

Comment:
Small nitpick, but I would add a `.spacedBy` here too. What you've done works great, but on small devices it might be hard to read as the text is shoved next to each other.